src/metabase/lib/column_group.cljc

```diff
@@ -39,59 +50,96 @@
   [_query _stage-number column-group]
   column-group)
 
+(defmulti ^:private display-info-for-group-method
```

Comment:
The `display-info-method` below got to be pretty big with the new stuff I added for joins to it so I broke it out into a private multimethod with a few separate impls. It's a little easier on the eyes this way. I didn't change the code for the other stuff, only for joins.